src/components/StartScriptGenerator.tsx

```diff
@@ -0,0 +1,152 @@
+    if (isServerSide) return;
+    const element = document.getElementById("output-command-text");
+    if (!element) return;
+
+    element.style.height = 'auto';
+    const scrollHeight = element.scrollHeight;
+
+    requestAnimationFrame(() => {
+        element.style.height = `${scrollHeight}px`;
+    });
+};
+
+const StartScriptGenerator: React.FC = () => {
+    const [memory, setMemory] = useState(4.0);
+    const [filename, setFilename] = useState('server.jar');
+    const [selectedFlag, setSelectedFlag] = useState(FLAGS.AIKARS);
+    const [dropdownVisible, setDropdownVisible] = useState(false);
+    const [guiEnabled, setGuiEnabled] = useState(false);
+    const dropdownRef = useRef(null);
+
+    const handleClickOutside = (event: { target: EventTarget | null }) => {
+        if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
+            setDropdownVisible(false);
+        }
+    };
+
+    useEffect(() => {
+        resizeOutput();
+        if (!isServerSide) return;
+        document.addEventListener('mousedown', handleClickOutside);
+
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [memory, filename, selectedFlag, guiEnabled]);
+
+    return (
+        <div className="server-config-container">
+            <h2>Server Configuration</h2>
+            <div className="config-section" style={{ width: "100%" }}>
+                <label id="memory-slider-label" className="sr-only">Memory Usage: {memory}GB</label>
+                <input
+                    id="memory-slider"
+                    type="range"
+                    min={0.5}
+                    max={24}
+                    step={0.5}
+                    value={memory}
+                    onChange={(e) => setMemory(parseFloat(e.target.value))}
+                    aria-labelledby="memory-slider-label"
+                />
+                <div className="slider-markers">
+                    {markerPoints.map((point) => (
+                        <div
+                            key={point}
+                            className="slider-marker"
+                            style={{ left: `${((point - 0.5) / 23.5) * 100}%` }}
+                        >
+                            {point}GB
+                        </div>
+                    ))}
+                </div>
+            </div>
+            <div className={"middle-flex-wrapper"}>
+                <div className="config-section" style={{ width: "40%" }}>
+                    <label htmlFor="filename-input">Filename:</label>
+                    <input
+                        id="filename-input"
+                        type="text"
+                        value={filename}
+                        onChange={(e) => setFilename(e.target.value)}
+                        className={"filename-input"}
+                    />
+                </div>
+                <div className="config-section" style={{ width: "40%" }}>
```

Comment:
remove all this inline style (and the 40% in the css) and give .middle-flex-wrapper a gap of 2rem and .config-section flex: 1, that achieves the same look but scales better